src/corva/api.py

```diff
@@ -113,22 +171,46 @@ def _request(
             **(headers or {}),
         }
 
-        response = requests.request(
-            method=method,
-            url=url,
-            params=params,
-            json=data,
-            headers=headers,
-            timeout=timeout,
-        )
+        if self.max_retries > 0:
+            retry_decorator = retry(
+                stop=stop_after_attempt(self.max_retries),
+                wait=wait_random_exponential(multiplier=0.25, max=10),
+                retry=retry_if_result(
+                    lambda r: r.status_code in retryable_status_codes
+                ),
+            )
+            retrying_request = retry_decorator(self._execute_request)
```

Comment:
Would retrying code block here be simpler? https://tenacity.readthedocs.io/en/latest/#retrying-code-block

e.g. there would be no need to add separate `_execute_request` method

Reply:
To be honest it was simpler before using 3-rd party retrying lib and the code we had initially was very similar to https://tenacity.readthedocs.io/en/latest/#retrying-code-block. I haven't managed make that code in example work without breaking anything.